feat: [TD-292] validations Note model

ross-garcia opened the pull request:
## Summary

Archivos modificados:
- modelo Note: se incluyen los métodos word_count, validate_content_length, invalid_content?, validate_content_length_error y content_length
- seed.rb: se añaden datos de prueba para la tabla notes
- locales/en.rb y locales/es.rb: mensajes de error custom para las validaciones del modelo Note
- Se elimina el test que verifica la relación que existe entre Note y User porque genera conflictos para ejecutar los test debido al modo en que está asociado con Utility
- modelo NorthUtility y  SouthUtility: métodos content_length_short y content_length_medium añadidos

## Screenshots

Tests:

![image](https://github.com/ross-garcia/Training-Rails-UGUITO-API-Bootstrap/assets/160248891/4d5ca775-1b62-458f-9223-1678639e2c8d)

## Test Cases
N/A

## Checklist 

- [+] Revisé los archivos modificados antes de liberar el pull request para revisión. 👩🏻‍💻
- [+] Actualicé la card de JIRA considerando: **estado**, **horas incurridas**, **enfoque tomado como comentario de la card** y **requisitos de configuración asociados**✅
- [+] El título de mi PR sigue la convención requerida 👉 [%scope - %descripcion](https://www.conventionalcommits.org/en/v1.0.0/)

## JIRA Card

https://widergy.atlassian.net/browse/TD-292

ross-garcia committed:
feat: validations Note model
- added word_count, content_length, content_length_of_south_utility and content_length_of_north_utility methods to Note model
- added seeds for Note (modified seed.rb file)
- custom message in locales files

JoaquinSeita commented on app/models/note.rb:
```diff
@@ -16,4 +16,39 @@ class Note < ApplicationRecord
+    content.split.length
+  end
+
+  def content_length_of_north_utility
+    case word_count
+    when 0..50
+      'short'
+    when 51..100
+      'medium'
+    else
+      'long'
+    end
+  end
+
+  def content_length_of_south_utility
+    case word_count
+    when 0..60
+      'short'
+    when 61..120
+      'medium'
+    else
+      'long'
+    end
+  end
+
+  def content_length
+    case utility.type
+    when 'SouthUtility'
+      content_lenght_of_south_utility
+    when 'NorthUtility'
+      content_lenght_of_north_utility
+    end
+  end
```
Algo bueno que tiene Ruby, es que facilita muchas cosas para evitar código repetido, tus metodos `content_length_of_north_utility` y `content_length_of_south_utility` son muy similares, podemos unificarlos de alguna forma
Por otro lado, que pasaria con esta logica si por ejemplo se suma otra Utility? El lunes vemos en una llamadita como podemos encarar estos problemas para dejar codigo que sea reutilizable

JoaquinSeita commented on config/locales/es.yml:
```diff
@@ -31,3 +31,16 @@
 
 es:
   hello: "Hola Mundo!"
+  activerecord:
+    errors:
+      models:
+        note:
+          attributes:
+            title:
+              blank: "%{attribute} es un campo requerido"
+            content:
+              blank: "%{attribute} es un campo requerido"
+            note_type:
+              blank: "%{attribute} es un campo requerido"
+            user_id:
+              blank: "%{attribute} debe incluir un ID de usuario válido"
```
En que momento se usan estos mensajes?

JoaquinSeita commented on db/seeds.rb:
```diff
@@ -31,4 +31,5 @@
 User.all.find_each do |user|
   random_books_amount = [1, 2, 3].sample
   FactoryBot.create_list(:book, random_books_amount, user: user, utility: user.utility)
+  FactoryBot.create(:note, title: 'This is a title', content: 'This is a content', note_type: Note.note_types.values.sample, user_id: user.id)
```
Excelente!

JoaquinSeita reviewed (commented):
Buen laburo Ross! Te dejo unos comentarios el lunes los vemos mas en profundidad :smiley:

ross-garcia committed:
feat: validations Note model
- NorthUtility and SouthUtility modified: content_length_short and content_length_medium methods added
- Note model modified: validate_content_length method added
- custom message added to validate_content_length
- content_length method modified

ross-garcia committed:
feat: test to verify the relationship with the User model deleted

JoaquinSeita commented on app/models/note.rb:
```diff
@@ -12,8 +12,33 @@
+  end
+
+  def invalid_content?
+    word_count > utility.content_length_short && review?
```
buenisimo ese `review?` :100:

JoaquinSeita reviewed (approved):
Se ve super para mi! :100:
